compiler/rustc_mir_build/src/build/expr/as_rvalue.rs

```diff
@@ -225,6 +228,36 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
                         }
                     })
                     .collect();
+
+                // For this closure introduce fake reads based on the the new data we wrote in
+                // typeck results. Note that the fake reads are introduced into MIR but not into
+                // the closure operands/ aggregate created.
+                //
+                // It's possible that we are creating a nested closure here and therefore weren't
+                // able to introduce fake reads in our parent, ensure that this an UpvarRef.
+                // Because at some point our parent probably introduced a fake read for this place.
+                debug!("RFC2229 = as_rvalue.rs ExprKind::Closure");
+                for hir_place in fake_reads.iter() {
+                    debug!("RFC2229 = as_rvalue.rs {:?}", hir_place);
+                    // Convert place into mir place
+                    let var = match hir_place.base {
+                        HirPlaceBase::Local(hir_id) => hir_id,
+                        HirPlaceBase::Upvar(upvar_id) => {
+                            debug!("upvar");
+                            upvar_id.var_path.hir_id
+                        },
+                        _ => {
+                            debug!("Do not know how to get HirId out of Rvalue and StaticItem");
+                            continue;
+                        },
+                    };
+                    debug!("RFC2229 = as_rvalue pushing following fake_read {:?}", var);
```

Comment:
you want to do something like `let place = unpack!(block = this.as_place(block, fake_read_place));`

where fake_read_place is a thir expr. and then use that mir place

Reply:
The current approach will not work in case of nested closures, because there u need a read of an Upvar and not a local variable to ur parent